pkg/apis/serving/v1beta1/trained_model.go

```diff
@@ -0,0 +1,33 @@
+package v1beta1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type TrainedModel struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              TrainedModelSpec   `json:"spec,omitempty"`
+	Status            TrainedModelStatus `json:"status,omitempty"`
+}
+type TrainedModelSpec struct {
+	// Required field for parent inference service
+	InferenceService string `json:"inferenceService"`
+	// Predictor model spec
+	PredictorModel ModelSpec `json:"predictorModel"`
+	// Explainer model spec
+	ExplainerModel ModelSpec `json:"explainerModel,omitempty"`
+}
```

Comment:
Is this a 1-of-n semantic? So you can just supply either a PredictorModel OR an ExplainerModel but not both? How does this match up to InferenceService which has both?

Maybe the best is to leave out ExplainerModel and you need to define it as now on the inference service so you can only have 1 model artifact with an explainer. The AlibiExplainerSpec already as a StorageURI in its spec.

Reply:
I can remove the ExplainerModel for now with the assumption that different predictors need to share the same explainer which is defined in the parent InferenceService.

Reply:
@cliveseldon actually there is routing problem, when you call `v1/models/model-a:explain` and `v1/models/model-b:explain` will explainer understand it is trying to reuse the same explainer model?

Reply:
True. At present the predictorUrl is injected as a single parameter. So you would need a solution to allow 1 explainer to call N models which are assumed to all behave similarly dependant on the request it receives. So looks like som refactoring of the present explainer python server is needed.

Reply:
I removed the ExplainerModel and I think we should also discuss wether Transformer and Explainer should be an independent CRD. In the world of multi-model inference, each model should be able to choose its transformer and explainer independently.